### PR TITLE
update libuv to 1.32.0

### LINKED ISSUE #374: add new api uv_tcp_close_reset
```
.. c:function:: int uv_tcp_close_reset(uv_tcp_t* handle, uv_close_cb close_cb)

    Resets a TCP connection by sending a RST packet. This is accomplished by
    setting the `SO_LINGER` socket option with a linger interval of zero and
    then calling :c:func:`uv_close`.
    Due to some platform inconsistencies, mixing of :c:func:`uv_shutdown` and
    :c:func:`uv_tcp_close_reset` calls is not allowed.

    .. versionadded:: 1.32.0
```

### PR DESCRIPTION
**Pull request by zhaozg**
close #374

### CODE REVIEW

**Comment by squeek502** [comment]
Failing tests on Linux/Windows:

Linux:
```
  { address = { ip = "127.0.0.1", port = 33627, family = "inet" }, server = uv_tcp_t: 0x1f88e90 }
  { req = uv_write_t: 0x1f9a048, socket = uv_tcp_t: 0x1f77dd0 }
  "client onwrite"	uv_tcp_t: 0x1f77dd0	nil
  "server read"	{ data = "Hello" }
  "client read"	{ data = "Hello" }
  "client read"	{  }
Uncaught Error: ./tests/test-tcp.lua:253: assertion failed!
stack traceback:
	[C]: in function 'assert'
	./tests/test-tcp.lua:253: in upvalue 'fn'
	./lib/tap.lua:48: in function <./lib/tap.lua:46>
	[C]: in function 'luv.run'
	./lib/tap.lua:55: in function <./lib/tap.lua:42>
	[C]: in function 'xpcall'
	./lib/tap.lua:42: in upvalue 'run'
	./lib/tap.lua:136: in function 'lib/tap'
	tests/run.lua:33: in main chunk
	[C]: in ?
UNCLOSED	uv_tcp_t: 0x1f77dd0
  ./lib/tap.lua:70: 1 unclosed handle
  stack traceback:
  	[C]: in function 'error'
  	./lib/tap.lua:70: in function <./lib/tap.lua:42>
  	[C]: in function 'xpcall'
  	./lib/tap.lua:42: in upvalue 'run'
  	./lib/tap.lua:136: in function 'lib/tap'
  	tests/run.lua:33: in main chunk
  	[C]: in ?
not ok 78 tcp - tcp close reset accepted
```
Windows:
```
  { server = uv_tcp_t: 0x037e3830, address = { ip = "127.0.0.1", family = "inet", port = 3110 } }
  { req = uv_write_t: 0x007ede30, socket = uv_tcp_t: 0x037e30a0 }
  "client onwrite"	uv_tcp_t: 0x037e30a0	nil
  "server read"	{ data = "Hello" }
  "client read"	{ err = "ECONNRESET" }
  .\lib/tap.lua:58: Missing 1 expected call
  stack traceback:
  	[C]: in function 'error'
  	.\lib/tap.lua:58: in function <.\lib/tap.lua:42>
  	[C]: in function 'xpcall'
  	.\lib/tap.lua:42: in function 'run'
  	.\lib/tap.lua:136: in function 'tap'
  	tests\run.lua:33: in main chunk
  	[C]: at 0x00835430
not ok 77 tcp - tcp close reset accepted
```

---

Note that there were also new API functions added in 1.31.0 that we are missing: [`uv_fs_statfs`](http://docs.libuv.org/en/v1.x/fs.html#c.uv_fs_statfs) and [`uv_os_environ`](http://docs.libuv.org/en/v1.x/misc.html#c.uv_os_environ). Those can be added in a separate PR, though.

**Comment by zhaozg** [comment]
@squeek502  Done, with three new api.

File: src/luv.c [hunk 1]
```diff
@@ -237,6 +240,9 @@ static const luaL_Reg luv_functions[] = {
   {"fs_readdir", luv_fs_readdir},
   {"fs_closedir", luv_fs_closedir},
 #endif
+#if LUV_UV_VERSION_GEQ(1, 28, 0)
```
Comment (squeek502) [inline src/luv.c]: This should be `LUV_UV_VERSION_GEQ(1, 31, 0)`

File: src/fs.c [hunk 2]
```diff
@@ -239,6 +239,9 @@ static int push_fs_result(lua_State* L, uv_fs_t* req) {
     case UV_FS_STAT:
     case UV_FS_LSTAT:
     case UV_FS_FSTAT:
+#if LUV_UV_VERSION_GEQ(1, 31, 0)
+    case UV_FS_STATFS:
```
Comment (squeek502) [inline src/fs.c]: Doesn't this need special handling? `statfs` gives a `uv_statfs_t`, not a `uv_stat_t` that `luv_push_stats_table` expects.

Comment (squeek502) [reply src/fs.c]: Also, `statfs`'s return is in `req->ptr` not `req->statbuf` [according to the docs](http://docs.libuv.org/en/v1.x/fs.html#c.uv_fs_statfs).

**Comment by zhaozg** [comment]
Done

**Comment by squeek502** [comment]
Oh, you also might want to update the version in CMakeLists.txt: https://github.com/luvit/luv/blob/master/CMakeLists.txt#L9-L11

**Comment by zhaozg** [comment]
> Oh, you also might want to update the version in CMakeLists.txt: https://github.com/luvit/luv/blob/master/CMakeLists.txt#L9-L11

Do that in https://github.com/luvit/luv/pull/377

**Pull request merged**